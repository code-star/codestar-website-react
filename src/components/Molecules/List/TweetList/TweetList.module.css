.tweetList {
	max-width: 500px;
}
