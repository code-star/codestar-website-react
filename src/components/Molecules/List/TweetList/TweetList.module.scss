.card {
  max-width: 345px;
  margin: 1em;
  display: flex;
  flex-direction: column;
}

.media {
  height: 0;
  padding-top: 56.25%; /* 16:9 */
}

a {
  color: black;

  &:hover {
    text-decoration: none;
  }

  .prefix {
    color: #888;
  }
}
