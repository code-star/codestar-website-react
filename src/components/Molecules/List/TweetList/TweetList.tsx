import React from 'react';
import { withStyles } from '@material-ui/core/styles';
import {
  Card,
  CardActionArea,
  CardContent,
  Typography,
  CardActions,
  Button,
} from '@material-ui/core';

interface ITweetListProps {
  tweets: any[];
}

const styles = {
  card: {
    maxWidth: 345,
  },
  media: {
    height: 140,
  },
};

const TweetList = ({ tweets }: ITweetListProps) => (
  <Card>
    <CardActionArea>
      <CardContent>
        <Typography gutterBottom variant="headline" component="h2">
          {tweets[0].user.name} tweets ...
        </Typography>
        {tweets.map(tweet => {
          return (
            <div key={tweet.id} className="p-2">
              <Typography component="p">
                <a
                  href={`https://twitter.com/${tweet.user.screen_name}/status/${
                    tweet.id_str
                  }`}
                >
                  <span className="text-info">{`${new Date(
                    tweet.created_at
                  ).toDateString()} `}</span>
                  {`${tweet.text}`}
                </a>
              </Typography>
            </div>
          );
        })}
      </CardContent>
    </CardActionArea>
    <CardActions>
      <Button
        size="small"
        color="primary"
        href={`https://twitter.com/${tweets[0].user.screen_name}?lang=en`}
      >
        Read More
      </Button>
    </CardActions>
  </Card>
);

export default withStyles(styles)(TweetList);
