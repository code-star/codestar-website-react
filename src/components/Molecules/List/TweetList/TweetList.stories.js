import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs } from '@storybook/addon-knobs';
import { Button } from '@material-ui/core';
import TweetList from './TweetList.tsx';
import tweets from '../../../../../public/mock/get-recent-tweets';

storiesOf('components/Molecules/List', module)
  .addDecorator(withKnobs)
  .add('TweetList', () => (
    <TweetList
      tweets={tweets.slice(0, 5)}
      eventDate="December 13, 2018"
      eventImage="https://secure.meetupstatic.com/photos/event/3/5/b/b/600_473173755.jpeg"
      eventName="Data Oriented Design with Maxim Zaks"
    >
      <Button size="small" color="primary" href="#">
        Sign Up
      </Button>
    </TweetList>
  ));
