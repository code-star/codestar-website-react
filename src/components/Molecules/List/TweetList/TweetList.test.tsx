import React from 'react';
import { shallow } from 'enzyme';

import TweetList from './TweetList';
import { CardContent } from '@material-ui/core';
import renderToJSON from '../../../../setupTests';

const someTweets = [
  {
    user: {
      name: 'someAwesomeName',
      screen_name: 'someAwesomeScreenName',
    },
    created_at: 'Fri Aug 31 13:51:31 +0000 2018',
    text: 'some awesome text',
    favorite_count: 3,
    retweet_count: 5,
    id: 1,
  },
  {
    user: {
      name: 'someOtherAwesomeName',
      screen_name: 'someOtherAwesomeScreenName',
    },
    created_at: 'Sun Sep 02 12:40:19 +0000 2018',
    text: 'some other awesome text',
    favorite_count: 1,
    retweet_count: 2,
    id: 2,
  },
];

const renderShallow = (tweets: any) => {
  return shallow(
    <TweetList tweets={tweets} eventImage="a" eventDate="b" eventName="c" />
  );
};

describe('<TweetList />', () => {
  let wrapper;

  describe('Instance', () => {
    it('must have an instance of CardContent', () => {
      wrapper = renderShallow(someTweets);
      expect(wrapper.find(CardContent)).toHaveLength(1);
    });
  });

  describe('Snaphot', () => {
    it('must match some tweet list', () => {
      expect(renderToJSON(renderShallow(someTweets).get(0))).toMatchSnapshot();
    });
  });
});
