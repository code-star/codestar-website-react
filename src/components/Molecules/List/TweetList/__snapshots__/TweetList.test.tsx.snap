// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TweetList /> Snaphot must match some tweet list 1`] = `
<div
  className="MuiPaper-root-2 MuiPaper-elevation8-12 MuiPaper-rounded-3 MuiCard-root-1"
>
  <div
    className="MuiCardHeader-root-29"
  >
    <div
      className="MuiCardHeader-avatar-30"
    >
      <div
        aria-label="Speaker Twitter Avatar"
        className="MuiAvatar-root-35"
      />
    </div>
    <div
      className="MuiCardHeader-content-32"
    >
      <span
        className="MuiTypography-root-38 MuiTypography-body2-46 MuiCardHeader-title-33"
      >
        someAwesomeName
      </span>
      <span
        className="MuiTypography-root-38 MuiTypography-body2-46 MuiTypography-colorTextSecondary-62 MuiCardHeader-subheader-34"
      >
        b
      </span>
    </div>
  </div>
  <div
    className="MuiCardMedia-root-64"
    src={undefined}
    style={
      Object {
        "backgroundImage": "url(\\"a\\")",
      }
    }
    title="c"
  />
  <div
    className="MuiCardContent-root-66"
  >
    <div
      className="p-2"
    >
      <a
        href="https://twitter.com/someAwesomeScreenName/status/undefined"
      >
        <div
          className="MuiTypography-root-38 MuiTypography-body1-47"
        >
          <img
            className="mr-2"
            src="https://res.cloudinary.com/codestar/image/upload/f_auto/w_30/v1532588516/codestar.nl/images/events/twitter"
          />
          Fri Aug 31 2018
        </div>
        <p
          className="MuiTypography-root-38 MuiTypography-body1-47"
        >
          some awesome text
        </p>
      </a>
    </div>
    <div
      className="p-2"
    >
      <a
        href="https://twitter.com/someOtherAwesomeScreenName/status/undefined"
      >
        <div
          className="MuiTypography-root-38 MuiTypography-body1-47"
        >
          <img
            className="mr-2"
            src="https://res.cloudinary.com/codestar/image/upload/f_auto/w_30/v1532588516/codestar.nl/images/events/twitter"
          />
          Sun Sep 02 2018
        </div>
        <p
          className="MuiTypography-root-38 MuiTypography-body1-47"
        >
          some other awesome text
        </p>
      </a>
    </div>
  </div>
  <div
    className="MuiCardActions-root-67"
  >
    <a
      className="MuiButtonBase-root-95 MuiButton-root-69 MuiButton-text-71 MuiButton-textSecondary-73 MuiButton-flat-74 MuiButton-flatSecondary-76 MuiButton-sizeSmall-92 MuiCardActions-action-68"
      href="https://twitter.com/someAwesomeScreenName?lang=en"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex="0"
    >
      <span
        className="MuiButton-label-70"
      >
        About 
        someAwesomeName
      </span>
      <span
        className="MuiTouchRipple-root-98"
      />
    </a>
  </div>
</div>
`;
